### PR TITLE
docs: describe supported Kubernetes versions

### PR DESCRIPTION
**Pull request by marquiz**
Fixes: #1257

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
Question for @ffromani @swatisehgal  do we have all we need for `Topology-updater` on Kubernetes 1.26?

**Comment by marquiz** [comment]
Changed from K8s v1.16 to v1.21 as that's the minimum version required by nfd-topology-updater. v1.20 is already so old (EOL'd nearly a year and a half ago) that didn't feel it necessary to describe minimum k8s version of different components

/unhold

**Comment by swatisehgal** [comment]
> > Question for @ffromani @swatisehgal do we have all we need for `Topology-updater` on Kubernetes 1.16?
> 
> Ha, good point @ArangoGutierrez. I totally forgot about topology-updater see_no_evil It doesn't work with k8s v1.16. When did the podresources improvement get beta? monocle_face I need to edit this pr /hold

Apologies for the delay in response, some PodResource API [improvements](https://github.com/kubernetes/kubernetes/pull/93243) were made in 1.20 and the GetallocatableResources endpoint was [introduced](https://github.com/kubernetes/kubernetes/pull/95734) as an alpha feature in 1.21 and [graduated to Beta](https://github.com/kubernetes/enhancements/pull/2932) in 1.23 so NFD-topology updater has everything it needs 1.21 onwards. In 1.21 and 1.22, feature gate needs to be enabled but is not needed from 1.23 onwards.

**Comment by marquiz** [comment]
@swatisehgal thanks. Just checked the docs and the [nfd-topology-updater](https://kubernetes-sigs.github.io/node-feature-discovery/v0.13/usage/nfd-topology-updater.html) page actually has detailed deployment notes (pre-1.23 must have the feature gate enabled). So I think v1.21+ on the intro page is good enough. I now added a bit more details on the usage/nfd-topology-update page

File: docs/deployment/index.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ sort: 2
 
 # Deployment
 
+Node Feature Discovery can be deployed on any recent version of Kubernetes
+(v1.21+).
```
Comment (ArangoGutierrez) [inline docs/deployment/index.md:10]: lgtm now

**Pull request merged**